### ISSUE TITLE
XML dependencies (e.g. images) not displayed in lens reader

### ISSUE THREAD

**Issue by mfelczak** [pkp/pkp-lib#3038]
When an XML galley is uploaded in OJS 3.1 along with dependent files (images, figures, etc.) the references to the files need to be rewritten to OJS-specific file paths (similar to HTML galleys).

**Comment by ajnyga** [comment]
I think I have that here along with some other changes: https://github.com/ajnyga/lensGalley

**Comment by asmecher** [comment]
@mfelczak, try https://github.com/asmecher/lensGalley/commit/7a32cbc8a64d431949728a833c5a5a4fa11fb535 and let me know if that resolves the issue.

@ajnyga, I didn't use your branch because it introduces changes into the third-party code, and we're trying to avoid that; instead, I used the same approach as exists in the HTML galleys, and adapted it a little for XML instead.

**Comment by ajnyga** [comment]
Sure, no problem. I realized now that I have seen those changes of yours, but did not remember them anymore.

I agree that it is good to stick with unchanged third-party code. But I would like to point out that the current lensGalley plugin uses a version of lens that is made specifically for eLife articles. This means that there are different kinds of small parsers that only work with specific kind of XML content. The whole idea of lens is to "build your own", so it would make sense that OJS would have one designed specifically for OJS. 

On our part I stopped working with lens when I realized that the new versions are not mobile friendly and came up with another solution for XML articles.
